internal/handlers/common/error.go

```diff
@@ -51,7 +51,41 @@ const (
 	ErrRegexMissingParen = ErrorCode(51091) // Location51091
 )
 
-// Error represents wire protocol error.
+// ProtoErr represents protocol error type.
+type ProtoErr interface {
```

Comment:
Why is this type introduced? I feel like I've seen `ProtoErr` somewhere else before... 
Would it be possible to document the reason / the use cases for this type to be here? E.g. where/when does one need to use this interface? 

The only usage I see is in the `ProtocolError` function which seems to be quite a generic wrapper. Why this PR is the first time this function was introduced? How should it be used in the future (e.g. when more operators are implemented)?

Reply:
> I need a WriteErr from https://github.com/FerretDB/FerretDB/pull/596 here, so added as a file.

It will be just copied from the other PR #596 as is, I just need to go further using that code.